Modules/FindCUDA/select_compute_arch.cmake

```diff
@@ -0,0 +1,176 @@
+# Borrowed from caffe
+# https://github.com/BVLC/caffe/blob/master/cmake/Cuda.cmake
+
+# Synopsis:
+# Function for selecting GPU arch flags for nvcc based on CUDA_ARCH_NAME
+# Usage:
+#   SELECT_NVCC_ARCH_FLAGS(out_variable)
+#
+# Variables affecting the choice:
+#
+# CUDA_ARCH_NAME: One of ("Fermi" "Kepler" "Kepler-M" "Maxwell" "Maxwell-M" "All" "Manual")
+# ENV{CUDA_ARCH_BIN} : set if CUDA_ARCH_NAME set to "Manual"
+#
+
+# Known NVIDIA GPU achitectures
+# This list will be used for CUDA_ARCH_NAME = All option
+
+if (NOT CUDA_arch_select_included)
```

Comment:
The right way to fix it is to export SELECT_NVCC_ARCH_FLAGS only to be visible by FindCUDA clients. I believe those should be left for the user to set and not set automatically.  How do I do that ?